text/0265-event-vision.md

```diff
@@ -0,0 +1,75 @@
+
+### OTLP
+
+Since OpenTelemetry Events are a type of OpenTelemetry Log, they share the same OTLP log data structure and pipeline.
+
+### API
+
+OpenTelemetry should provide a (user-facing) Log API that includes the capability to emit OpenTelemetry Events.
+
+### Interoperability with Generic Logging Libraries
+
+It should be possible to send OpenTelemetry Logs from the OpenTelemetry Log API to a generic logging library (e.g., Log4j).
+This allows users to integrate OpenTelemetry Logs into an existing (non-OpenTelemetry) log stream.
+
+It should also be possible to bypass the OpenTelemetry Log API entirely and emit OpenTelemetry Logs (including Events)
+directly via existing language-specific logging libraries, if that library has the capability to do so.
+
+OpenTelemetry will recommend that
+[instrumentation libraries](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/glossary.md#instrumentation-library)
+use the OpenTelemetry Log API to emit OpenTelemetry Events rather than using a generic logging library. This recommendation aims to provide users with a simple and consistent
+onboarding experience that avoids mixing approaches.
+
+OpenTelemetry will also recommend that application developers use the OpenTelemetry Log API to emit OpenTelemetry Events instead of using a generic
+logging library, as this helps prevent accidentally emitting logs that lack an event name or are unstructured.
+
+Recommending the OpenTelemetry Log API for emitting OpenTelemetry Events, rather than using a generic logging library, contributes to a clearer overall
+OpenTelemetry API story. This ensures a unified approach with first-class user-facing APIs for traces, metrics, and events,
+all suitable for direct use in native instrumentation.
+
+### Relationship to Span Events
+
+Events are intended to replace Span Events in the long term.
+Span Events will be deprecated to signal that users should prefer Events.
+
+Interoperability between Events and Span Events will be defined in the short term.
+
+### SDK
+
+This refers to the existing OpenTelemetry Log SDK.
+
+## Trade-offs and mitigations
+
+TODO
+
+## Prior art and alternatives
+
+TODO
+
+## Open questions
+
+* How to support routing logs from the Log API to a language-specific logging library
+  while simultaneously routing logs from the language-specific logging library to an OpenTelemetry Logging Exporter?
+* How do log bodies interoperate with generic logging libraries?
+* How do event bodies interoperate with Span Events?
+
+## Future possibilities
+
+* The Event API will likely need an `IsEnabled` function based on severity level, scope name, and event name.
+* Ergonomic improvements to make it more attractive as a replacement for generic logging APIs.
+* Capturing raw metric events as opposed to aggregating and emitting them as OpenTelemetry Metric data
+  (e.g. [opentelemetry-specification/617](https://github.com/open-telemetry/opentelemetry-specification/issues/617)).
```

Comment:
This sounds wrong to me. From instrumentation perspective, I think we want users to stick with an API rather than switching among APIs based on how they want to consume the data. For example, there are cases where folks want both aggregated metrics and raw measurements - in this case we do not want them to instrument twice with metrics and logging APIs.

Reply:
I'd prefer to be overgenerous in adding open questions / future possibilities (and not worry about them until we agree on the basic vision).

Reply:
(btw this came from @jsuereth and I don't think it is suggesting to instrument twice but rather to emit raw events from behind the metric API)

Reply:
If it is open question, 💯 agree. I just want to avoid folks reading this as if it is the direction that we agreed.

Reply:
it's under the "future possibilities" section, I'm interpreting that to be a _possibility_, not something we have agreed on, so you think that needs more clarification in the text?

Reply:
I might be confused by "open questions" and "future possibilities" - I thought "open questions" are something we haven't decided, "future possibilities" are something that we generally agreed on but haven't funded due to capacity reasons.

Reply:
got it, I've moved everything to "open questions" to avoid any ambiguity